Editor/CoreAPI/StateManager.cs

```diff
@@ -49,6 +49,7 @@ public string BucketName
             {
                 _selectedProfile.BucketName = value;
                 SaveProfiles();
+                CoreApi.PutSetting(SettingsKeys.CurrentBucketName, value);
```

Comment:
Similarly, I see we're setting the SettingsKeys.CurrentRegion above. Is that for the same reason?

Reply:
`CurrentRegion` seems to be used in:
- `RegionBootstrap.Refresh`
- `RegionBootstrap.Save`
- `DeploymentRequestFactory.CreateRequest`
- `BootstrapUtility.GetBootstrapData`

These would all have to be refactored